skstats/hypotests/calculators/basecalculator.py

```diff
@@ -0,0 +1,371 @@
+            poinull = POI(mean, [1.1, 1.2, 1.0])
+            poialt = POI(mean, 1.2)
+            nll = calc.expected_pvalue([poinull], [poialt])
+        """
+        self.checkpois(poinull)
+        if poialt:
+            self.checkpois(poialt)
+            self.checkpoiscompatibility(poinull, poialt)
+
+        return self._expected_pvalue_(poinull=poinull, poialt=poialt, nsigma=nsigma, CLs=CLs, qtilde=qtilde,
+                                      onesided=onesided, onesideddiscovery=onesideddiscovery)
+
+    def _expected_pvalue_(self, poinull, poialt, nsigma, CLs, qtilde, onesided, onesideddiscovery):
+        raise NotImplementedError
+
+    def expected_poi(self, poinull: List[POI], poialt: List[POI], nsigma, alpha=0.05, CLs=False,
+                     onesided=True, onesideddiscovery=False):
+        """Computes the expected parameter of interest values such that the expected p_values == $$\alpha$$
+        for different values of $$\\sigma$$ (0=expected/median)
+
+        Args:
+            poinull (List[`hypotests.POI`]): parameters of interest for the null hypothesis
+            poialt (List[`hypotests.POI`], optionnal): parameters of interest for the alternative hypothesis
+            nsigma (`numpy.array`): array of values of $$\\sigma$$ to compute the expected pvalue
+            CLs (bool, optionnal): if `True` uses pvalues as $$p_{cls}=p_{null}/p_{alt}=p_{clsb}/p_{clb}$$
+                else as $$p_{clsb} = p_{null}$
+            onesided (bool, optionnal): if `True` (default) computes onesided pvalues
+            onesideddiscovery (bool, optionnal): if `True` (default) computes onesided pvalues for a discovery
+
+        Returns:
+            `numpy.array`: array of expected POI values for each $$\\sigma$$ value
+
+        Example:
+            mean = zfit.Parameter("mu", 1.2)
+            poinull = POI(mean, [1.1, 1.2, 1.0])
+            poialt = POI(mean, 1.2)
+            nll = calc.expected_poi([poinull], [poialt])
+        """
+        self.checkpois(poinull)
+        if poialt:
+            self.checkpois(poialt)
+            self.checkpoiscompatibility(poinull, poialt)
+
+        return self._expected_poi_(poinull=poinull, poialt=poialt, nsigma=nsigma, alpha=alpha, CLs=CLs,
+                                   onesided=onesided, onesideddiscovery=onesideddiscovery)
+
+    def _expected_poi_(self, poinull, poialt, nsigma, alpha, CLs, onesided, onesideddiscovery):
+        raise NotImplementedError
+
+    @staticmethod
+    def checkpois(pois):
+        msg = "A list of POIs is required."
+        if not isinstance(pois, (list, tuple)):
+            raise ValueError(msg)
+        if not all(isinstance(p, POI) for p in pois):
+            raise ValueError(msg)
+        if len(pois) > 1:
+            msg = "Tests with more that one parameter of interest are not yet implemented."
+            raise NotImplementedError(msg)
+
+    @staticmethod
+    def checkpoiscompatibility(poi1, poi2):
```

Comment:
Add some "_" here ;-)?

Reply:
So check_poiscompatibility ?

Reply:
I would even write `check_pois_compatibility` in this instance.